### DISCUSSION TITLE
Firebird dialect

### DISCUSSION THREAD

**Discussion by dquinter** [MoonStorm/FastCrud#158] [Ideas]
Hi all.

Can I use FastCRUD with Firebird or any other unsupported database engine? Or how can I implement that dialect? What points do I have to observe to implement it?

Thank you so much!

**Comment by MoonStorm** [comment]
Sorry for the late reply. Dialects live in [Dapper.FastCrud/SqlBuilders/Dialects](https://github.com/MoonStorm/Dapper.FastCRUD/tree/master/Dapper.FastCrud/SqlBuilders/Dialects). 
The factory method is in  [Dapper.FastCrud/EntityDescriptor/EntityDescriptor(TEntity).cs](https://github.com/MoonStorm/Dapper.FastCRUD/blob/master/Dapper.FastCrud/EntityDescriptors/EntityDescriptor(TEntity).cs).
We do need the dialect to be covered by tests though, even if that means running non-automated tests against a simple local database. You can observe the different categories we have in the Dapper.FastCrud.Tests project.
